Integration tests passing in CI, failing locally

## Description
In my most recent PR (#71) the integration tests in CI passed, but when I ran them locally, I got some failures.

```
...
        assert 'error_id' in response_data
        assert 'description' in response_data
        assert 'timestamp' in response_data
        assert 'context' in response_data
    
        assert response_data['http_code'] == status_code
>       assert response_data['error_id'] == error_id
E       AssertionError

tests/utils.py:46: AssertionError
======================================================= 13 failed, 107 passed in 31.85 seconds ========================================================
ERROR: InvocationError: '/Users/edaniszewski/dev/vaporio/synse-server/.tox/py36/bin/pytest -s --junitxml=results/pytest/junit.xml --cov-report html:results/cov-html --cov-report term --cov=synse --html=results/pytest/index.html --self-contained-html --profile tests/integration'
_______________________________________________________________________ summary _______________________________________________________________________
ERROR:   py36: commands failed
```

I need to
- (a) figure out why it passes on CI but fails remotely
- (b) figure out why its failing
- (c) fix the failures

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -8,7 +8,7 @@ import shutil
 import pytest
 import yaml
 
-from synse import config, factory
+from synse import config, const, factory
 from tests import data_dir
 
 _app = None
@@ -18,6 +18,14 @@ _app = None
 def app():
     """Fixture to get a Synse Server application instance."""
     global _app
+
+    # override the default config directory location for testing. this is
+    # to prevent collision with anything in the local default socket directory,
+    # which could be the case if you are running plugins locally directly
+    # on the host (e.g. for development)
+    const.SOCKET_DIR = os.path.join(data_dir, 'socks')
+
+    # if the app doesn't exist, create it
     if _app is None:
         with open(os.path.join(data_dir, 'config.yml'), 'w') as f:
             yaml.dump({'log': 'debug'}, f)

